### ISSUE TITLE
Parser gap: $(( command substitution starting with a subshell is parsed as arithmetic expansion

### ISSUE THREAD

**Issue by ss-o** [z-shell/zsh-lint#266]
## Summary

`$((` is ambiguous: it can open an arithmetic expansion (`$(( expr ))`) or a command substitution whose first word is a subshell (`$( (list) ... )`). Native Zsh resolves this per input; `x=$((echo hi) | cat)` runs as a command substitution and sets `x` to `hi`. zsh-lint's front end commits to the arithmetic reading unconditionally for any `$((`, so this form is rejected.

Out of scope: `x=$((1 2))` fails at runtime under native Zsh as well as zsh-lint, with `` bad math expression: operator expected at `2' ``. That is a genuine arithmetic error, not this ambiguity, and is not part of this issue.

## Minimized script

```zsh
x=$((echo hi) | cat)
```

This is mvdan/sh's own canonical example for this ambiguity (see References).

## `zsh -f -n` on that script

passes (no output, exit 0)

A second, runtime oracle confirms which reading Zsh actually applies: `zsh -f -c 'x=$((echo hi) | cat); print -r -- "[$x]"'` prints `[hi]`, exit 0, so Zsh executes this as a command substitution, not as an arithmetic expansion.

## zsh-lint output

```
probe-2_pipe.zsh:1:11: not a valid arithmetic operator: `hi`
```

## Mechanism

mvdan/sh's lexer tokenizes `$((` as a single `dollDblParen` token purely lexically, with no lookahead into the content that follows and no backtracking. Confirmed directly in the `mvdan.cc/sh/v3@v3.14.1` module source (`syntax/lexer.go`, two identical blocks around lines 562-567 and 707-712):

```go
case '(':
    if p.rune() == '(' {
        p.rune()
        return dollDblParen
    }
    return dollParen
```

There is nothing between the check and the `return`; the token is produced before any content is read. That token then drives `ArithmExp` construction in `syntax/parser.go` (~lines 1240-1263): `p.followArithm` / `p.arithmEnd` parse the content as an arithmetic expression unconditionally and report a parse error if it is not one. There is no retry of the same span as a `CmdSubst` containing a `Subshell`.

Confirmed general, not specific to any one inner construct: `x=$((echo hi) )` (a bare space before the final `)`, no pipe) and `x=$(() { :; } arg)` (the anonymous-function-invocation shape encountered while investigating #265) both run natively the same way, producing `hi` and an empty string respectively, and both are rejected by zsh-lint with the same class of arithmetic-parse error. `x=$((echo hi))`, where the closing `))` is adjacent, is a different case: Zsh itself evaluates it as arithmetic and fails at runtime, matching zsh-lint's rejection, so it does not belong in this issue's repro set.

mvdan/sh's own upstream issue tracker has a closed issue on exactly this ambiguity, [mvdan/sh#30](https://github.com/mvdan/sh/issues/30) (2016). mvdan/sh originally added backtracking for it, then removed it, citing this text from the [POSIX Shell Command Language spec](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_06_03): "A conforming application shall ensure that it separates the \"$(\" and '(' into two tokens (that is, separate them with white space) in a command substitution that starts with a subshell." The maintainer's own words on removing it: "people should not rely on the `$((` ambiguity anyway." Upstream declined to keep the ambiguity supported; a fix has to live on this repository's side. The adapter layer's existing byte-rewrite-and-reparse pattern fits, since the resulting tree is the correct one (for example `parseAnonymousFunctionArgs` masking bytes and retrying, `parseMultiNameFor` rewriting and rebasing positions via `rebaseForError`).

## Classification

zshexpn, Arithmetic Expansion (`$((...))`, "ARITHMETIC EXPANSION" section), versus zshexpn, Command Substitution (the `$(...)` form containing a parenthesized subshell as its first word, "COMMAND SUBSTITUTION" section). Confirmed both section headers live in `man zshexpn`, not `zshmisc`.

## Where it occurs

Found manually while investigating #265 (probing whether the same-line anonymous-invocation gap also affected the `$(...)` command-substitution form); not yet confirmed against the survey corpus (`docs/project/corpus.md`).

## zsh-lint version

`main` @ `6f06274`. Also reproduces on PR #264's branch `bug-255` @ `5fa4f49`.

## Acceptance criteria

- A `gap-<id>-dollar-dbl-paren-ambiguity.zsh` fixture matching the minimized script fails on the current parser and passes both `zsh -f -n` and a runtime check confirming the command-substitution reading.
- Content following `$((` that does not parse as a valid arithmetic expression, when Zsh's own disambiguation would read it as a command substitution containing a subshell, is accepted, matching native Zsh, without breaking genuine `$(( expr ))` arithmetic parsing.
- `x=$( (echo hi) )` (already passing, unambiguous form) and other existing arithmetic-expansion tests continue to pass unchanged.
- `x=$((1 2))` and other inputs that are genuinely invalid arithmetic under native Zsh's own runtime evaluation are unaffected by this fix; they remain a separate, correctly-diagnosed rejection, not part of this gap.

## References

- #265: a different, unrelated gap found investigating the same repro family (anonymous invocation inside a same-line backtick substitution)
- #233: another arithmetic-expression parser gap, same `ArithmExp` surface, different cause (unsupported math function calls inside the arithmetic grammar itself, affecting both `((...))` and `$((...))`)
- [mvdan/sh#30](https://github.com/mvdan/sh/issues/30): the same ambiguity in the upstream parser; backtracking was added and then removed, citing the POSIX text quoted above
